stake/packages/frontend/lib/contracts.ts

```diff
@@ -0,0 +1,89 @@
+import type { ContractData } from '@hoprnet/hopr-ethereum'
+import { chainIdToNetwork, chainIdToEnvironmentId } from './connectors'
+
+export type Contracts = {
+  [key in
+    | 'wxHoprToken'
+    | 'xHoprToken'
+    | 'hoprBoost'
+    | 'hoprStake']: ContractData
+}
+
+// cache results
+let last_chainId = 0
+let contracts: Contracts | undefined = undefined
+
+export default async function getContracts(
+  chainId: number
+): Promise<typeof contracts> {
+  if (contracts && last_chainId === chainId) return contracts
+
+  const network = chainIdToNetwork(chainId)
+  const envId = chainIdToEnvironmentId(chainId)
+
+  const [hoprToken, hoprBoost, hoprStake] = await Promise.all([
+    import(
+      `@hoprnet/hopr-ethereum/deployments/${envId}/${network}/HoprToken.json`
+    ).then((res) => res.default),
+    import(
+      `@hoprnet/hopr-ethereum/deployments/${envId}/${network}/HoprBoost.json`
+    ).then((res) => res.default),
+    import(
+      `@hoprnet/hopr-ethereum/deployments/${envId}/${network}/HoprStake.json`
+    ).then((res) => res.default),
+  ])
+
+  const wxHoprToken: ContractData = {
+    ...(envId === 'master-goerli' ? hoprToken : xDai_wxHOPR),
+    abi: ERC677_ABI,
+    transactionHash: '', // not required by this website
+  }
+
+  const xHoprToken: ContractData = {
+    ...(envId === 'master-goerli' ? goerli_xHOPR : xDai_xHOPR),
+    abi: ERC677_ABI,
+    transactionHash: '', // not required by this website
+  }
+
+  last_chainId = chainId
+  contracts = {
+    wxHoprToken,
+    xHoprToken,
+    hoprBoost,
+    hoprStake,
+  }
+
+  return contracts
+}
+
+// TODO: read from protocol-config
```

Comment:
Is this going to be fixed in this PR as well?

Reply:
not on this PR - we don't export `protocol-config` by itself, would like us first to finish our decoupling work before doing so